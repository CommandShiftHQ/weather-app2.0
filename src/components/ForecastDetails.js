import React from "react";
import PropTypes from "prop-types";
import Moment from "moment";
import WeatherIcon from "react-icons-weather";

const ForecastDetails = ({ forecast }) => {
  const { date, icon, temperature, wind, humidity } = forecast;
  return (
    <>
      <div className="big-date">
        <span>{Moment(date).format("ddd Do MMM")}</span>
      </div>
      <br />
      <div className="big-icon">
        <span>
          {
            <WeatherIcon
              name="owm"
              iconId={icon}
              flip="horizontal"
              rotate="90"
            />
          }
        </span>
      </div>

      <div className="temp-max">
        <span>Max Temperature: {temperature.max}&deg;c</span>
      </div>
      <br />
      <div className="temp-min">
        <span>Min Temperature: {temperature.min}&deg;c</span>
      </div>
      <br />
      <div className="big-humidity">
        <span>Humidity: {humidity}%</span>
      </div>
      <br />
      <div className="wind-speed">
        <span>
          Wind: {wind.speed}mph {<i className={"wi wi-towards-nne"} />}
        </span>
      </div>
    </>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.object.isRequired,
};

export default ForecastDetails;
