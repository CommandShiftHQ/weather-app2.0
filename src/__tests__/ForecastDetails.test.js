import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const mockForecast = {
    date: 1590418800000,
    humidity: 32,
    icon: "800",
    temperature: { max: 22, min: 21 },
    wind: { speed: 4, direction: "wsw" },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={mockForecast} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct city and location props", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails forecast={mockForecast} />
    );

    expect(getByText("Mon 25th May")).toHaveClass("big-date");
    expect(getByTestId("test-icon")).toHaveClass("big-icon");
    expect(getByText("Max Temperature: 22°c")).toHaveClass("temp-max");
    expect(getByText("Min Temperature: 21°c")).toHaveClass("temp-min");
    expect(getByText("Humidity: 32%")).toHaveClass("big-humidity");
    expect(getByText("Wind: 4mph")).toHaveClass("wind-speed");
  });
});
