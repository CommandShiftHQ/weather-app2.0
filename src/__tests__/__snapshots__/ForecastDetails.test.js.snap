// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="big-date"
  >
    Mon 25th May
  </div>
  <br />
  <div
    class="big-icon"
    data-testid="test-icon"
  >
    <i
      class="wi wi-day-sunny wi-flip-horizontal wi-rotate-90"
      name="owm"
    />
  </div>
  <div
    class="temp-max"
  >
    Max Temperature: 22°c
  </div>
  <br />
  <div
    class="temp-min"
  >
    Min Temperature: 21°c
  </div>
  <br />
  <div
    class="big-humidity"
  >
    Humidity: 32%
  </div>
  <br />
  <div
    class="wind-speed"
  >
    Wind: 4mph 
    <i
      class="wi wi-towards-nne"
    />
  </div>
</DocumentFragment>
`;
